lib/services/local/plugins.go

```diff
@@ -32,11 +32,15 @@ const (
 
 // PluginsService manages plugin instances in the backend.
 type PluginsService struct {
-	backend backend.Backend
+	getBackend func() backend.Backend
```

Comment:
why do you need to use it as a function?

Reply:
This is an established pattern used in Enterprise/Cloud addons, where the initialization of these addons happens before the Auth (and its backend) is initialized fully.

See the Enterprise PR: https://github.com/gravitational/teleport.e/pull/844/files#diff-d4ea8e373e1f92f375929cfe3e9b2e4cb93700829fe7fef7689c552a0add6f3aR230

and an existing example of such code: https://github.com/gravitational/teleport.e/blob/82681e1ce14ad1ae321a8d31b4baf8e1b8f15e46/lib/auth/plugin.go#L129

Reply:
Got it